src/pydvl/value/shapley/truncated.py

```diff
@@ -196,55 +229,57 @@ def truncated_montecarlo_shapley(
         sampling permutations.
     :param truncation: callable that decides whether to stop computing
         marginals for a given permutation.
-    :param n_jobs: number of jobs processing permutations. If None, it will be
-        set to :func:`available_cpus`.
     :param config: Object configuring parallel computation, with cluster
         address, number of cpus, etc.
+    :param n_jobs: Number of permutation monte carlo jobs
+        to run concurrently.
     :param coordinator_update_period: in seconds. How often to check the
         accumulated results from the workers for convergence.
     :param worker_update_period: interval in seconds between different
         updates to and from the coordinator
     :return: Object with the data values.
 
     """
-    # Avoid circular imports
-    from .actor import get_shapley_coordinator, get_shapley_worker
-
-    if config.backend == "sequential":
-        raise NotImplementedError(
-            "Truncated MonteCarlo Shapley does not work with "
-            "the Sequential parallel backend."
-        )
-
-    coordinator = get_shapley_coordinator(config=config, done=done)  # type: ignore
-
-    workers = [
-        get_shapley_worker(  # type: ignore
-            u,
-            coordinator=coordinator,
-            truncation=truncation,
-            worker_id=worker_id,
-            update_period=worker_update_period,
-            config=config,
-        )
-        for worker_id in range(effective_n_jobs(n_jobs, config=config))
-    ]
-    for worker in workers:
-        worker.run(block=False)
-
-    while not coordinator.check_convergence():
-        sleep(coordinator_update_period)
-
-    return coordinator.accumulate()
-
-    # Something like this would be nicer, but it doesn't seem to be possible
-    # to start the workers from the coordinator.
-    # coordinator.add_workers(
-    #     n_workers=n_jobs,
-    #     u=u_id,
-    #     update_period=worker_update_period,
-    #     config=config,
-    #     truncation=truncation,
-    # )
-    #
-    # return coordinator.run(delay=coordinator_update_period)
+    n_jobs = effective_n_jobs(n_jobs, config)
+
+    algorithm = "truncated_montecarlo_shapley"
+
+    parallel_backend = init_parallel_backend(config)
+    u = parallel_backend.put(u)
```

Comment:
Isn't it ugly, revealing that one must "put" some data? Ideally this would be hidden from a user of the parallel backend

Reply:
This is the best I could come up with without tying the executor to the specific parallel backend.